src/EVM/Transaction.hs

```diff
@@ -47,25 +53,63 @@ data Transaction = Transaction {
     txAccessList :: [AccessListEntry],
     txMaxPriorityFeeGas :: Maybe W256,
     txMaxFeePerGas :: Maybe W256
-} deriving Show
+} deriving (Show, Generic)
+instance JSON.ToJSON Transaction where
+  toJSON t = JSON.object [ ("input",             (JSON.toJSON $ t.txData))
+                         , ("gas",               (JSON.toJSON $ "0x" ++ showHex (toInteger $ t.txGasLimit) ""))
+                         , ("gasPrice",          (JSON.toJSON $ show $ fromJust $ t.txGasPrice))
+                         , ("v",                 (JSON.toJSON $ show $ (t.txV)-27))
+                         , ("r",                 (JSON.toJSON $ show $ t.txR))
+                         , ("s",                 (JSON.toJSON $ show $ t.txS))
+                         , ("to",                (JSON.toJSON $ t.txToAddr))
+                         , ("nonce",             (JSON.toJSON $ show $ t.txNonce))
+                         , ("value",             (JSON.toJSON $ show $ t.txValue))
+                         , ("type",              (JSON.toJSON $ t.txType))
+                         , ("accessList",        (JSON.toJSON $ t.txAccessList))
+                         , ("maxPriorityFeePerGas", (JSON.toJSON $ show $ fromJust $ t.txMaxPriorityFeeGas))
+                         , ("maxFeePerGas",      (JSON.toJSON $ show $ fromJust $ t.txMaxFeePerGas))
+                         , ("chainId",           (JSON.toJSON ("0x1" :: String))) -- NOTE: should be part of struct?
```

Comment:
what does this comment mean?

Reply:
The field `chainID` is not part of the Transaction datatype. It's the only element needed for this datatype to be 1-to-1 usable in `evmtool` as a transaction. I wrote this comment to remind myself that I should discuss with someone whether this should be made part of the datatype or if it's OK as-is, added to the `ToJSON` but not actually be part of the datatype.

What do you think? I think in many (or only some?) places in HEVM, chainID is assumed to be `1`. So I thought maybe it's OK to simply sneak this in there.

Reply:
Hmmm... actually, many places where `Transaction` is being used, `chainId` also needs to be provided. So maybe we could make it part of the datatype? what do you think?

Reply:
Yep, much cleaner, actually! I managed to remove a bunch of parameters being passed around

Reply:
@d-xo Let's discuss about making it a `Maybe`?

Reply:
So it's staying in as-is as discussed, and we'll make it a Maybe in case it needs to be one. I'm resolving this because we agreed on this already!